src/App.tsx

```diff
@@ -2,49 +1,111 @@
 import 'bulma/css/bulma.css';
+import { useState } from 'react';
+import classNames from 'classnames';
+
 import './App.scss';
+import { goodsFromServer } from './api/goodsFromServer';
+import { SortType } from './enums/sortingEnum';
+
+export const App = () => {
+  const [goods, setGoods] = useState(goodsFromServer);
+  const [activeSort, setActiveSort] = useState<SortType | ''>('');
+  const [isReversed, setIsReversed] = useState(false);
+
+  const sortGoods = (compareFunction: (a: string, b: string) => number) => {
+    const sortedGoods = [...goods].sort(compareFunction);
+
+    setGoods(sortedGoods);
+  };
+
+  const sort = (typeOfSortGoods: SortType) => {
+    let compareFunction;
+
+    switch (typeOfSortGoods) {
+      case SortType.Alphabetically:
+        compareFunction = (a: string, b: string) => {
+          return isReversed ? b.localeCompare(a) : a.localeCompare(b);
+        };
+
+        setActiveSort(SortType.Alphabetically);
+        break;
+
+      case SortType.ByLength:
+        compareFunction = (a: string, b: string) => {
+          return isReversed ? b.length - a.length : a.length - b.length;
+        };
+
+        setActiveSort(SortType.ByLength);
+        break;
+
+      default:
+        throw new Error(`Unknown sort type: ${typeOfSortGoods}`);
+    }
+
+    sortGoods(compareFunction);
+  };
+
+  const reverseGoods = () => {
+    setGoods(prevGoods => [...prevGoods].reverse());
+    setIsReversed(prevIsReversed => !prevIsReversed);
+  };
+
+  const resetGoods = () => {
+    setGoods([...goodsFromServer]);
+    setIsReversed(false);
+    setActiveSort('');
```

Comment:
The function's name 'resetGoods' does not start with a verb and does not clearly indicate what it does. According to the checklist, function names should start with a verb. Consider renaming it to something like 'handleResetGoods' to make its purpose clearer.